src/components/LinkEncoder.js

```diff
@@ -91,29 +99,50 @@ const LinkEncoder = () => {
 
     const [buttonText, setButtonText] = useState('Connect');
     const [selected, setSelected] = useState("success");
+    const [checked, setIsChecked] = useState(false);
 
     const connectAndDisconnect = async (e) => {
-        e.preventDefault();
-
-        await window.linkEncoderAPI.connectionLinkEncoder(postData.ip, postData.port);
+        
+
+        await window.linkEncoderAPI.connectionLinkEncoder(postData.ip, postData.port)
+        .then(() => {
+            if (buttonText == 'Connect'){
+                setButtonText('Disconnect');
+                setSelected("error");
+            }
+            else {
+                setButtonText('Connect');
+                setSelected("success");
+            }  
+        });
     };
 
-    const handleClick = () => {
-        if (buttonText == 'Connect'){
-            setButtonText('Disconnect');
-            setSelected("error");
-        }
-        else {
-            setButtonText('Connect');
-            setSelected("success");
-        }   
+    const pinging = async () => {
+        return new Promise(resolve => {
+            if (checked == false) {
```

Comment:
It's because when the check button is clicked, it will change the state into false not true. This is what I see in the tutorials for React checkboxes. Not sure if it can be change, but I can look into it.